containers/src/Kokkos_ScatterView.hpp

```diff
@@ -1365,31 +1397,41 @@ create_scatter_view(Op, Duplication, Contribution,
 namespace Kokkos {
 namespace Experimental {
 
-template <typename DT1, typename DT2, typename LY, typename ES, typename OP,
+
+template <typename DT1, typename DT2, typename LY, typename DV, typename ES,
+          typename OP, typename CT, typename DP, typename... VP>
+void contribute(ES const& exec_space,
```

Comment:
Templating over the execution space distinct from the device of the views being worked on may not be how you'd like to do this. I think it's safe/sane, though, since the `contribute_into` call then constrains it to exactly the right type. I'm very open to suggestions for how to change it.

Reply:
I think this is resolved, now that I specify the argument in terms of the typedef inside `ES` as originally defined.